### ISSUE TITLE
IQueryable as client collection

### ISSUE THREAD

**Issue by Adiel-Sharabi** [object/Simple.OData.Client#271]
Hi, 
Is there any way to return IQueryable so the paging/filtering etc will apply directly on the server? 
Thanks!

**Comment by object** [comment]
Hi,
OData protocol supports paging/filtering via query strings included in the request URI, and Simple.OData.Client implements it in its Skip/Top/Filter methods. Actually the library only parses the response on the client side, it never filter any data that are coming from the server.

**Comment by Adiel-Sharabi** [comment]
The IQueryable interface give power full tool to 3rd party control (GridView etc) to make Linq query against the source in common way. The actual filter in applied on the server via the URI and standard OData. To implement this we need an IQueryable provider that "know" how to translate the Linq based expression to the underline provider. 
I tried to implement it but it's not so simple. The MS Odata client contain methods that return IQueryable and it's work's very well. 
I'm wondering you intend to implement it (for me it's deal breaker since we using controls that relay on it) 
BTW - very nice work on this library! 
Thanks.

**Comment by object** [comment]
While the Simple.OData.Client API look similar to LINQ, it differs from LINQ and this was a conscious choice that I tried to explain in this artilce (http://www.codeproject.com/Articles/686240/reasons-to-consume-OData-feeds-using-Simple-ODa). Microsoft WCF Data Service client took LINQ-base approach, and it often confused developers who believe they can use all handy LINQ methods only to encounter runtime errors later because of lack of support from OData protocol.
Simple.OData.Client tries to stay close the OData protocol, so its users need to understand how OData works. This is why it has methods Top and Filter rather than Take and Where.

I used to support Simple.Data OData adapter but not anymore - also for the reason of providing an interface that will be closer to OData features rather than offering one-fits-all solution that might work better for traditional SQL database but wouldn't fit OData protocol.

So if you absolutely must have LINQ, then WCF Data Client is the way to go - with all unpleasant discoveries that will be occur only at runtime. Otherwise it's a deliberate choice to use an API that closely reflects OData capabilities but doesn't offer traditional LINQ features.
